neural_compressor/torch/quantization/algorithm_entry.py

```diff
@@ -40,32 +40,12 @@ def rtn_entry(
     model: torch.nn.Module, configs_mapping: Dict[Tuple[str, callable], RTNConfig], *args, **kwargs
 ) -> torch.nn.Module:
     """The main entry to apply rtn quantization."""
-    from neural_compressor.torch.algorithms.weight_only import rtn_quantize
+    from neural_compressor.torch.algorithms.weight_only import RTNQuantizer
 
-    # rebuild weight_config for rtn_quantize function
-    weight_config = {}
-    for (op_name, op_type), quant_config in configs_mapping.items():
-        if quant_config.name != RTN:
-            continue
-        weight_config[op_name] = {
-            "dtype": quant_config.dtype,
-            "bits": quant_config.bits,
-            "scheme": "sym" if quant_config.use_sym else "asym",
-            "group_size": quant_config.group_size,
-            "group_dim": quant_config.group_dim,
-            "use_full_range": quant_config.use_full_range,
-            "use_mse_search": quant_config.use_mse_search,
-            "use_layer_wise": quant_config.use_layer_wise,
-            "export_compressed_model": quant_config.export_compressed_model,
-            "use_double_quant": quant_config.use_double_quant,
-            "double_quant_dtype": quant_config.double_quant_dtype,
-            "double_quant_bits": quant_config.double_quant_bits,
-            "double_quant_scheme": "sym" if quant_config.double_quant_use_sym else "asym",
-            "double_quant_group_size": quant_config.double_quant_group_size,
-        }
-
-    model = rtn_quantize(model, weight_config=weight_config)
-    return model
+    logger.info("Quantize model with the RTN algorithm.")
+    algo = RTNQuantizer(configs_mapping)
```

Comment:
Here we need to clarify whether the algorithm should allow users to construct the "config_mapping" designed by INC or allow users to construct a simple dictionary. @ftian1